olive/passes/onnx/pipeline/__init__.py

```diff
@@ -0,0 +1,74 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import onnx
+
+TENSOR_TYPE_MAP = {
+    "float": onnx.TensorProto.FLOAT,
+    "uint8": onnx.TensorProto.UINT8,
+    "int8": onnx.TensorProto.INT8,
+    "uint16": onnx.TensorProto.UINT16,
+    "int16": onnx.TensorProto.INT16,
+    "int32": onnx.TensorProto.INT32,
+    "int64": onnx.TensorProto.INT64,
+    "string": onnx.TensorProto.STRING,
+    "bool": onnx.TensorProto.BOOL,
+    "float16": onnx.TensorProto.FLOAT16,
+    "double": onnx.TensorProto.DOUBLE,
+    "uint32": onnx.TensorProto.UINT32,
+    "uint64": onnx.TensorProto.UINT64,
+    "complex64": onnx.TensorProto.COMPLEX64,
+    "complex128": onnx.TensorProto.COMPLEX128,
+    "bfloat16": onnx.TensorProto.BFLOAT16,
+}
+
+
+def resolve_placeholder(model: onnx.ModelProto, param_value: dict):
+    resolved_param = param_value
+    arg_type = param_value.get("type")
+    if arg_type == "__model_input__":
```

Comment:
The code looks a bit duplicated.

Reply:
the parameter name is different. one is input_index and another is output_index